src/main/java/com/atlassian/bitbucket/jenkins/internal/samples/BitbucketSamples.java

```diff
@@ -0,0 +1,103 @@
+package com.atlassian.bitbucket.jenkins.internal.samples;
+
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.google.inject.Injector;
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.cps.GroovySample;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.UUID;
+
+public class BitbucketSamples {
+
+    private BitbucketSamples() {
+        //this class is a holder for samples. As such it should never be instantiated
+    }
+
+    @Extension
+    public static class BitbucketMavenSample implements GroovySample {
+
+        @Inject
+        private Injector injector;
+
+        public BitbucketMavenSample() {
+            //required by Jenkins
+        }
+
+        public BitbucketMavenSample(Injector injector) {
+            //Used to to ease testing
+            this.injector = injector;
+        }
+
+        @Override
+        public String name() {
+            return "bbs-maven";
+        }
+
+        @Override
+        public String script() {
+            if (injector != null) {
+                BitbucketPluginConfiguration configuration = injector.getInstance(BitbucketPluginConfiguration.class);
+
+                try {
+                    StringBuffer output = new StringBuffer();
+                    Files.readAllLines(Paths.get(BitbucketMavenSample.class.getResource("/samples/bbsMaven.groovy").toURI())).forEach(line -> {
+                        if (line.endsWith("{replace}")) {
+                            output.append("                ");
+                            List<BitbucketServerConfiguration> serverList = configuration.getValidServerList();
+                            if (serverList.size() == 0) {
+                                output.append("\n//NOTE! You need to configure a Bitbucket server at the global level to use Bitbucket Server");
```

Comment:
We can add a dummy output for sure, I thought it was more valuable to say 'hey you need to do this thing before this will even work' rather than pretend it may work. Thoughts?

Reply:
Yes on reflection I agree. They can't use the snippet and there's no other way to get the server ID, so the flow would be "set up server first, then look at snippet"